Client/Client.Qt/Widgets/ChatHistory.cpp

```diff
@@ -17,37 +19,78 @@ ChatHistory::ChatHistory(QWidget* parent) : QWidget(parent), _messageList()
-    auto msg = new MessageWidget(history, message, _messageId++, _userId, utc, user);
+    if(reply != nullptr)
+    {
+        auto replyMsg = new ReplyMessageWidget(history, reply->getMessage(), reply->getMessageId(), reply->getUsername());
+        auto msg = new MessageWidget(history, message, _messageId++, _userId, utc, user);
+        replyMsg->show();
+        replyMsg->resize(history->width() - 25, replyMsg->expectedHeight());
+        msg->show();
+        msg->resize(history->width() - 25, msg->expectedHeight());
+
+        history->setMinimumHeight(history->minimumHeight() + replyMsg->expectedHeight() + 10);
+
+        connect(replyMsg, &ReplyMessageWidget::geometryChanged, this, [=](int diff) {
+            history->setMinimumHeight(history->minimumHeight() + diff);
+            updateLayout(true);
+        });
+
+
+        history->setMinimumHeight(history->minimumHeight() + msg->expectedHeight() + 10);
+
+
+        connect(msg, &MessageWidget::geometryChanged, this, [=](int diff) {
+            history->setMinimumHeight(history->minimumHeight() + diff);
+            updateLayout(true);
+        });
+
+        connect(msg, &MessageWidget::createReplySignal, this, &ChatHistory::createReplySignal);
+
+        _messageAndReplyList.push_back(std::make_pair(std::unique_ptr<MessageWidget>(nullptr), std::unique_ptr<ReplyMessageWidget>(replyMsg)));
+        _messageAndReplyList.push_back(std::make_pair(std::unique_ptr<MessageWidget>(msg), std::unique_ptr<ReplyMessageWidget>(nullptr)));
+
+        updateLayout();
+
+        _scrollArea->scrollToWidget(msg);
+
+        _replyCount++;
+
+        reply->close();
+    }
+    else
+    {
+        auto msg = new MessageWidget(history, message, _messageId++, _userId, utc, user);
+        msg->show();
+        msg->resize(history->width() - 25, msg->expectedHeight());
+        _messageAndReplyList.push_back(std::make_pair(std::unique_ptr<MessageWidget>(msg), std::unique_ptr<ReplyMessageWidget>(nullptr)));
 
-    msg->show();
-    msg->resize(history->width() - 25, msg->expectedHeight());
-    history->setMinimumHeight(history->minimumHeight() + msg->expectedHeight() + 10);
+         history->setMinimumHeight(history->minimumHeight() + msg->expectedHeight() + 10);
 
-    _messageList.push_back(std::unique_ptr<MessageWidget>(msg));
+        connect(msg, &MessageWidget::geometryChanged, this, [=](int diff) {
+            history->setMinimumHeight(history->minimumHeight() + diff);
+            updateLayout(true);
+        });
 
-    connect(msg, &MessageWidget::geometryChanged, this, [=](int diff) {
-        history->setMinimumHeight(history->minimumHeight() + diff);
-        updateLayout(true);
-    });
+        connect(msg, &MessageWidget::createReplySignal, this, &ChatHistory::createReplySignal);
 
-    updateLayout();
+        updateLayout();
 
-    _scrollArea->scrollToWidget(msg);
+        _scrollArea->scrollToWidget(msg);
+    }
 
-    messageAdded();
+    emit messageAdded();
```

Comment:
is emit that necessary?

Reply:
I think if we are using signals, then it is probably better to use emit. it's just easier to understand what it is